includes/Core/REST_API/REST_Routes.php

```diff
@@ -654,4 +661,38 @@ private function parse_google_response_data( $data ) {
 
 		return $data;
 	}
+
+	/**
+	 * Converts a WP_Error to its response representation.
+	 *
+	 * Adapted from \WP_REST_Server::error_to_response
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param WP_Error $error Error to transform.
+	 *
+	 * @return array
+	 */
+	protected function error_to_response( WP_Error $error ) {
+		$errors = array();
+
+		foreach ( (array) $error->errors as $code => $messages ) {
+			foreach ( (array) $messages as $message ) {
+				$errors[] = array(
+					'code'    => $code,
+					'message' => $message,
+					'data'    => $error->get_error_data( $code ),
+				);
+			}
+		}
+
+		$data = $errors[0];
+		if ( count( $errors ) > 1 ) {
+			// Remove the primary error.
+			array_shift( $errors );
+			$data['additional_errors'] = $errors;
+		}
+
+		return $data;
```

Comment:
See above: Although not super nice, we could potentially add a flag here that this _is_ in fact a `WP_Error`, for JavaScript.

Reply:
I wouldn't mind adding such a flag except that it wouldn't apply to non-batch responses, but the handler is used for both.

Reply:
I think if the behavior involved were more critical, this would be justified. Right now a false positive would likely only add a warning in the console. It's very unlikely that a false positive would trigger the auth/permission notices.